x-pack/plugins/reporting/server/export_types/csv_from_savedobject/lib/get_csv_job.ts

```diff
@@ -75,6 +77,14 @@ export const getGenerateCsvParams = async (
     fields: indexPatternFields,
   } = indexPatternSavedObject;
 
+  if (!indexPatternFields || indexPatternFields.length === 0) {
```

Comment:
logging addition for the open issues with Download CSV from a dashboard panel (to be fixed in https://github.com/elastic/kibana/pull/88303)